src/coreComponents/physicsSolvers/fluidFlow/FlowSolverBase.cpp

```diff
@@ -491,6 +487,97 @@ void FlowSolverBase::precomputeData( MeshLevel & mesh,
   }
 }
 
+void FlowSolverBase::initialize( DomainPartition & domain )
+{
+  // Compute hydrostatic equilibrium in the regions for which corresponding field specification tag has been specified
+  computeHydrostaticEquilibrium( domain );
+
+  forDiscretizationOnMeshTargets( domain.getMeshBodies(), [&]( string const &,
+                                                               MeshLevel & mesh,
+                                                               arrayView1d< string const > const & regionNames )
+  {
+    initializePorosityAndPermeability( mesh, regionNames );
+    initializeHydraulicAperture( mesh, regionNames );
+
+    // Initialize primary variables from applied initial conditions
+    initializeFluid( mesh, regionNames );
+
+    // Initialize the rock thermal quantities: conductivity and solid internal energy
+    // Note:
+    // - This must be called after updatePorosityAndPermeability and updatePhaseVolumeFraction
+    // - This step depends on porosity and phaseVolFraction
+    if( m_isThermal )
+    {
+      initializeThermal( mesh, regionNames );
+    }
+
+    // Save initial pressure and temperature fields
+    saveInitialPressureAndTemperature( mesh, regionNames );
+  } );
+
+  // report to the user if some pore volumes are very small
+  // note: this function is here because: 1) porosity has been initialized and 2) NTG has been applied
+  validatePoreVolumes( domain );
+}
+
+void FlowSolverBase::initializePorosityAndPermeability( MeshLevel & mesh, arrayView1d< string const > const & regionNames )
+{
+  // Update porosity and permeability
+  // In addition, to avoid multiplying permeability/porosity bay netToGross in the assembly kernel, we do it once and for all here
+  mesh.getElemManager().forElementSubRegions< CellElementSubRegion, SurfaceElementSubRegion >( regionNames, [&]( localIndex const,
+                                                                                                                 auto & subRegion )
+  {
+    // Apply netToGross to reference porosity and horizontal permeability
+    CoupledSolidBase const & porousSolid =
+      getConstitutiveModel< CoupledSolidBase >( subRegion, subRegion.template getReference< string >( viewKeyStruct::solidNamesString() ) );
+    PermeabilityBase const & permeability =
+      getConstitutiveModel< PermeabilityBase >( subRegion, subRegion.template getReference< string >( viewKeyStruct::permeabilityNamesString() ) );
+    arrayView1d< real64 const > const netToGross = subRegion.template getField< fields::flow::netToGross >();
+    porousSolid.scaleReferencePorosity( netToGross );
+    permeability.scaleHorizontalPermeability( netToGross );
+
+    // in some initializeState versions it uses newPorosity, so let's run updatePorosityAndPermeability to compute something
```

Comment:
this double update is still confusing but i don't really want to touch the constitutives...